test/gtest/log_test_helper.hpp

```diff
@@ -0,0 +1,170 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "miopen/logger.hpp"
+#include <gtest/gtest.h>
+#include <miopen/convolution.hpp>
+
+namespace Env {
+
+void setEnvironmentVariable(const std::string& name, const std::string& value)
+{
+    int ret = 0;
+
+#ifdef _WIN32
+    std::string env_var(name + "=" + value);
+    ret = _putenv(env_var.c_str());
+#else
+    ret = setenv(name.c_str(), value.c_str(), 1);
+#endif
+    EXPECT_EQ(ret, 0);
+}
+
+void unSetEnvironmentVariable(const std::string& name)
+{
+    int ret = 0;
+#ifdef _WIN32
+    std::string empty_env_var(name + "=");
+    ret = _putenv(empty_env_var.c_str());
+#else
+    ret = unsetenv(name.c_str());
+#endif
+    EXPECT_EQ(ret, 0);
+}
+
+} // namespace Env
+
+// Captures the std::cerr buffer and store it to a string.
+struct CerrRedirect
+{
+public:
+    CerrRedirect()
+    {
+        // Store the current stream buffer object and set
+        // stringstream as a new stream buffer.
+        old_stream_buf = std::cerr.rdbuf(str_buffer.rdbuf());
+        EXPECT_TRUE(old_stream_buf != nullptr);
+    }
+
+    std::string getString() { return str_buffer.str(); }
+
+    ~CerrRedirect()
+    {
+        // reset the old stream buffer back
+        std::cerr.rdbuf(old_stream_buf);
+    }
+
+private:
+    std::streambuf* old_stream_buf;
+    std::stringstream str_buffer;
+};
+
+struct Tensor
+{
+    miopenTensorDescriptor_t desc{};
+    void* data;
+    size_t data_size;
+
+    Tensor(int n, int c, int h, int w)
+    {
+        EXPECT_EQ(miopenCreateTensorDescriptor(&desc), 0);
+        EXPECT_EQ(miopenSet4dTensorDescriptor(desc, miopenFloat, n, c, h, w), 0);
+        data_size = n * c * h * w * sizeof(float);
+        EXPECT_EQ(hipMalloc(&data, data_size), hipSuccess);
+    }
+
+    ~Tensor()
+    {
+        miopenDestroyTensorDescriptor(desc);
+        hipFree(data);
```

Comment:
Try this. Otherwise the OpenCL test will complain. 
#if MIOPEN_BACKEND_OPENCL
            clReleaseMemObject(data);
#elif MIOPEN_BACKEND_HIP
            hipFree(data);
#endif